RiotNSE/NotificationService.swift

```diff
@@ -225,28 +231,37 @@ class NotificationService: UNNotificationServiceExtension {
         contentHandler?(content)
     }
     
-    func notificationContent(forEvent event: MXEvent, inAccount account: MXKAccount, onComplete: @escaping (UNNotificationContent?) -> Void) {
+    func notificationContent(forEvent event: MXEvent, inSession session: MXSession, onComplete: @escaping (UNNotificationContent?) -> Void) {
         guard let content = event.content, content.count > 0 else {
             NSLog("[NotificationService][Push] notificationContentForEvent: empty event content")
             onComplete(nil)
             return
         }
         
-        guard let room = account.mxSession.room(withRoomId: event.roomId) else {
+        guard let room = MXRoom(roomId: event.roomId, matrixSession: session, andStore: store) else {
             NSLog("[NotificationService][Push] notificationBodyForEvent: Unknown room")
             onComplete(nil)
             return
         }
+        
         let pushRule = room.getRoomPushRule()
         
-        room.state({ (roomState: MXRoomState!) in
-            
+        //  initialize a temporary file store, just to load the room state
+        let fileStore = MXFileStore(credentials: session.credentials)
```

Comment:
Why not using the existing `NSEMemoryStore `and implement `stateOfRoom` there?

Reply:
Yes, totally makes sense. Changing.